buffer/src/test/java/io/netty/buffer/PooledByteBufAllocatorTest.java

```diff
@@ -24,20 +26,140 @@
 import java.util.List;
 import java.util.Queue;
 import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.LockSupport;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class PooledByteBufAllocatorTest {
 
-    private static final int[] ALLOCATION_SIZES = new int[16 * 1024];
-    static {
-        for (int i = 0; i < ALLOCATION_SIZES.length; i++) {
-            ALLOCATION_SIZES[i] = i;
+    // The ThreadDeathWatcher sleeps 1s, give it double that time.
+    @Test (timeout = 2000)
+    public void testThreadCacheDestroyedByThreadDeathWatcher() {
+        int numArenas = 11;
+        final PooledByteBufAllocator allocator =
+            new PooledByteBufAllocator(numArenas, numArenas, 8192, 1);
+
+        for (int i = 0; i < numArenas; i++) {
+            new FastThreadLocalThread(new Runnable() {
+                @Override
+                public void run() {
+                    ByteBuf buf = allocator.newHeapBuffer(1024, 1024);
+                    for (int i = 0; i < buf.capacity(); i++) {
+                        buf.writeByte(0);
+                    }
+
+                    assertTrue(allocator.numThreadLocalCaches() > 0);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2186)